crates/node/src/lib.rs

```diff
@@ -380,6 +375,21 @@ impl Node {
             address = %my_node_config.address,
             "My Node ID is {}", my_node_config.current_generation);
 
+        // establish connection to admin nodes to announce the updated NodesConfiguration
+        // (presence of this node)
+        for admin_node in nodes_config.get_admin_nodes() {
+            // we don't have to announce us to ourselves
+            if admin_node.current_generation != my_node_id {
+                if let Err(err) = self
+                    .networking
+                    .node_connection(admin_node.current_generation)
+                    .await
+                {
+                    info!("Failed connecting to admin node '{}'. This can indicate network problems: {err}", admin_node.current_generation)
+                }
+            }
+        }
```

Comment:
What if we have one of the admin servers down? Is this strictly necessary or could controller periodically check nodes config from metadata store?

If this is a temporary thing (maybe until gossip), let's leave a todo describing your future plan please.

Reply:
Yes, this is a stop-gap solution for letting the cluster controller know that there is a newer `NodesConfiguration`. Additionally, the cluster controller will also periodically sync the `NodesConfiguration` which is implemented in #2058.

Until now, we relied on the attach call from the `PartitionProcessorManager`. However, if a node does not run the `Worker` role, then this won't happen. I'll add a comment explaining that this is a temporary solution until we have the gossipping mechanism. I'll also move the logic into a background task so that we don't block the node on connecting successfully to the admin nodes.